### PR TITLE
feat(wallet): Add Transfero On-Ramp Provider

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/26882

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
 -  TBA

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_constants.cc [hunk 1]
```diff
@@ -261,6 +283,9 @@ const std::vector<mojom::OnRampCurrency>& GetOnRampCurrenciesList() {
       {"GBP",
        "British Pound Sterling",
        {mojom::OnRampProvider::kWyre, mojom::OnRampProvider::kRamp}},
+      {"BRL",
+       "Brazilian Real(Transfero only)",
```
Comment (kdenhartog) [inline components/brave_wallet/browser/brave_wallet_constants.cc:287]:
```suggestion
       "Brazilian Real (Transfero only)",
```

File: components/brave_wallet/browser/brave_wallet_constants.h [hunk 2]
```diff
@@ -36,6 +36,8 @@ constexpr char kRampID[] = "8yxja8782as5essk2myz3bmh4az6gpq4nte9n2gf";
 constexpr char kSardineStorefrontBaseURL[] = "https://crypto.sardine.ai";
 constexpr char kSardineClientTokensURL[] =
     "https://api.sardine.ai/v1/auth/client-tokens";
+constexpr char kTransferoURL[] =
+    "https://app-onramp-development.azurewebsites.net";
```
Comment (kdenhartog) [inline components/brave_wallet/browser/brave_wallet_constants.h:40]: This looks like the incorrect URL for this

File: components/brave_wallet/common/brave_wallet.mojom [hunk 3]
```diff
@@ -519,6 +519,7 @@ enum OnRampProvider {
   kWyre = 0,
   kRamp = 1,
   kSardine = 2,
+  kTransfero = 3,
```
Comment (kdenhartog) [inline components/brave_wallet/common/brave_wallet.mojom:522]: This collides with the value for Transak: https://github.com/brave/brave-core/pull/15948/files#diff-9f2cf11d57c8efee5461b92cdb38fa327b70c25fd8d009f7692049e8e88944a8R528

File: components/brave_wallet_ui/stories/locale.ts [hunk 4]
```diff
@@ -403,12 +403,15 @@ provideStrings({
   braveWalletBuyRampNetworkName: 'Ramp.Network',
   braveWalletBuyWyreName: 'Wyre',
   braveWalletBuySardineName: 'Sardine',
+  braveWalletBuyTransferoName: 'Transfero',
   braveWalletBuyRampDescription: 'Buy with CC/Debit or ACH. Competitive Rates.',
   braveWalletBuyWyreDescription: 'Buy with CC/Debit or ACH. Competitive Rates.',
   braveWalletBuySardineDescription: 'Instant buy with your bank account. Lower fees.',
+  braveWalletBuyTransferoDescription: 'Placeholder description to be updated.',
```
Comment (kdenhartog) [inline components/brave_wallet_ui/stories/locale.ts:410]: NIT: update this

File: components/resources/wallet_strings.grdp [hunk 5]
```diff
@@ -481,12 +481,15 @@
...
+  <message name="IDS_BRAVE_WALLET_BUY_WITH_TRANSFERO" desc="Buy with Transfero option label">Buy with Transfero</message>
   <message name="IDS_BRAVE_WALLET_BUY_RAMP_NETWORK_NAME" desc="Ramp.Network onramp provider name">Ramp.Network</message>
   <message name="IDS_BRAVE_WALLET_BUY_WYRE_NAME" desc="Wyre onramp provider name">Wyre</message>
   <message name="IDS_BRAVE_WALLET_BUY_SARDINE_NAME" desc="Sardine onramp provider name">Sardine</message>
+  <message name="IDS_BRAVE_WALLET_BUY_TRANSFERO_NAME" desc="Transfero onramp provider name">Transfero</message>
   <message name="IDS_BRAVE_WALLET_BUY_RAMP_DESCRIPTION" desc="Ramp.Network description in buy options list">Buy crypto with Visa or Mastercard.</message>
   <message name="IDS_BRAVE_WALLET_BUY_WYRE_DESCRIPTION" desc="Wyre description in buy options list">Buy crypto with Visa or Mastercard.</message>
   <message name="IDS_BRAVE_WALLET_BUY_SARDINE_DESCRIPTION" desc="Sardine description in buy options list">Instant buy with your bank account. Lower fees.</message>
+  <message name="IDS_BRAVE_WALLET_BUY_TRANSFERO_DESCRIPTION" desc="Transfero description in buy options list">Placeholder description to be updated.</message>
```
Comment (kdenhartog) [inline components/resources/wallet_strings.grdp:492]: nit: update this

**Review by kdenhartog — COMMENTED** [review]
few early comments for review while it's still in draft

File: components/brave_wallet/browser/brave_wallet_constants.h (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet/browser/brave_wallet_constants.h:42]: @kdenhartog This url will change once Transfero complete their URL migration which currently ongoing. This is the main reason this PR is in draft.

File: components/resources/wallet_strings.grdp (hunk 5 above)
Comment (muliswilliam) [reply components/resources/wallet_strings.grdp:509]: See above

File: components/brave_wallet/browser/brave_wallet_constants.h (hunk 2 above)
Comment (kdenhartog) [reply components/brave_wallet/browser/brave_wallet_constants.h:42]: All good, that's what I figured.

**Comment by kdenhartog** [comment]
Overall this is looking pretty good. Once we've got the API url (and presumably the API auth method) coded up I'll retake a look. I'm also going to wait and see about the convo linked in the security review to see if that's something we want to do as apart of this or if consensus is to hold off for now. Overall what's here right now looks good though.

**Comment by muliswilliam** [comment]
@kdenhartog Great. Transfero does not need authentication just an Api Key appended to the URL. This change will come when the URL is ready as mentioned above.

**Pull request closed without merging**